sklearn/utils/estimator_checks.py

```diff
@@ -283,6 +283,25 @@ def check_estimator(Estimator):
     estimator : estimator object or class
         Estimator to check. Estimator is a class object or instance.
 
+    generate_only : bool, optional (default=True)
+        When `True`, checks are evaluated when `check_estimator` is called.
+        When `False`, `check_estimator` generates the checks and the estimator.
+
+    Notes
+    -----
+    Setting `generate_only=False` allows for checks to be disaseembled when
+    testing an estimator. To use this with pytest this can be done with:
+
+    .. code-block:: python
+
+        from itertools import chain
+
+        @pytest.mark.parameterize(
+            'estimator, check',
+            chain.from_iterable(check_estimator(est, generate_only=False)
+                                for est in estimators))
```

Comment:
One issues with this is that because `estimator` is an instance and not a class it's not going to be rendered nicely in the list of tests. That's why we use `id=_rename_partial`. I guess there is no easy way of fixing this for users...

Reply:
Without our `_rename_partial`, pytest names everything `estimator213-check41`. It could be slightly better if we yield the name as well from `check_estimator`, which will result in `ARDRegression-estimator211-check21`.

Reply:
I think we should really try to have the tests have sensible names. This is super useless otherwise I think.

Reply:
Why not fix the fixme in ``_rename_partial`` and move it to ``estimator_checks`` and make it public and use it here? That solves the problem, right?

Reply:
Using `print_changed_only` adds newlines sometimes, which leads to pytest names with newlines in them.

Making `_rename_partial` is a good idea, it would need another name, maybe `nicer_check_estimator_id`? 🤔

Reply:
how about using ``print_changed_only`` and remove the newlines? Also: if you're testing stuff that's that deeply nested there's no way to give it a good name.